src/util/recentlyEditedTree.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'lodash'
+import { timestamp } from './timestamp'
+import { pathToContext } from './pathToContext'
+import { timeDifference } from '../util.js'
+
+const editTimeMax = 30 // time diff limit in second for replacing descendants by ancestor
+
+const findTreeDeepestSubcontext = (tree, path) => {
```

Comment:
I recommend adding a jsdoc style comment above function with a description of what it does. Even with descriptive names, this is a nice convention that will help other developers and allow for auto-generated docs in the future.

Reply:
Sure.